app/controllers/admin/petition_details_controller.rb

```diff
@@ -0,0 +1,30 @@
+class Admin::PetitionDetailsController < Admin::AdminController
+  respond_to :html
+  before_action :find_petition
+
+  def show
+    if @petition.in_todo_list?
```

Comment:
Could also just change the ``find_petition`` method to use ``Petition.todo_list.find(params[:petition_id])``